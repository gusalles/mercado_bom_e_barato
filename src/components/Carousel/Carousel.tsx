import { Pagination, Autoplay } from 'swiper/modules';
import {
  StyledCarousel,
  Slide,
  SlideImage,
  VignetteFilter,
} from './Carousel.types';

import 'swiper/css';
import 'swiper/css/navigation';
import 'swiper/css/pagination';

interface CarouselProps {
  images: string[];
  alt: string;
}

export default function Carousel({ images, alt }: CarouselProps) {
  return (
    <StyledCarousel
      loop
      pagination
      autoplay={{ delay: 3000 }}
      modules={[Pagination, Autoplay]}
    >
      {images.map((image: string, index: number) => (
        <Slide key={index}>
          <SlideImage src={image} alt={alt} fill />
          <VignetteFilter />
        </Slide>
      ))}
    </StyledCarousel>
  );
}
