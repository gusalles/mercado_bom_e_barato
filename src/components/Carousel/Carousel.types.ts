import { styled } from '@mui/material';
import Image from 'next/image';
import { Swiper, SwiperProps, SwiperSlide } from 'swiper/react';

export const StyledCarousel = styled(Swiper)<SwiperProps>(({ theme }) => ({
  position: 'relative',
  width: '100%',
  height: 300,
  borderRadius: 4,
  [theme.breakpoints.down('sm')]: {
    borderRadius: 0,
    height: 150,
  },
}));

export const Slide = styled(SwiperSlide)(() => ({
  position: 'relative',
  width: '100%',
  height: '100%',
}));

export const SlideImage = styled(Image)(() => ({
  objectFit: 'contain',
}));

export const VignetteFilter = styled('span')(() => ({
  position: 'absolute',
  inset: 0,
  background: `radial-gradient(ellipse at center, rgba(0,0,0,0) 60%, rgba(0, 0, 0, 0.082) 100%)`,
  pointerEvents: 'none',
}));
