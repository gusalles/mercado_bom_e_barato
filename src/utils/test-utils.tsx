import { render } from '@testing-library/react';
import { ReactElement } from 'react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

export * from '@testing-library/react';

export function renderWithClient(ui: ReactElement) {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  });

  return render(
    <QueryClientProvider client={queryClient}>{ui}</QueryClientProvider>
  );
}
