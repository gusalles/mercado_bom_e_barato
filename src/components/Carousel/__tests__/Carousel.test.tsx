import { render, screen } from '@/utils/test-utils';
import { Carousel } from '../Carousel';
import {
  firstProduct,
  secondProduct,
} from '@/containers/Product/__mocks__/product.mocks';

describe('Carousel', () => {
  it('should render', () => {
    render(
      <Carousel
        images={[firstProduct.thumbnail, secondProduct.thumbnail]}
        alt="image alt"
      />
    );

    const carousel = screen.getByTestId('carousel');

    expect(carousel).toMatchSnapshot();
  });
});
