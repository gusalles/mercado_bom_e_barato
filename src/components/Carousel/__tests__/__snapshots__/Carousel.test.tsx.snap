// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Carousel > should render 1`] = `
<div
  class="swiper swiper-initialized swiper-horizontal css-13ddiq9"
  data-testid="carousel"
>
  <div
    class="swiper-wrapper"
    style="transition-duration: 0ms; transition-delay: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
  >
    <div
      class="swiper-slide css-1f8sh1y"
      data-swiper-slide-index="0"
    >
      <img
        alt="image alt"
        class="css-g98gbd"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=https%3A%2F%2Fcdn.dummyjson.com%2Fproduct-images%2Fbeauty%2Fessence-mascara-lash-princess%2Fthumbnail.webp&w=3840&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fcdn.dummyjson.com%2Fproduct-images%2Fbeauty%2Fessence-mascara-lash-princess%2Fthumbnail.webp&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fcdn.dummyjson.com%2Fproduct-images%2Fbeauty%2Fessence-mascara-lash-princess%2Fthumbnail.webp&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fcdn.dummyjson.com%2Fproduct-images%2Fbeauty%2Fessence-mascara-lash-princess%2Fthumbnail.webp&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fcdn.dummyjson.com%2Fproduct-images%2Fbeauty%2Fessence-mascara-lash-princess%2Fthumbnail.webp&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fcdn.dummyjson.com%2Fproduct-images%2Fbeauty%2Fessence-mascara-lash-princess%2Fthumbnail.webp&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fcdn.dummyjson.com%2Fproduct-images%2Fbeauty%2Fessence-mascara-lash-princess%2Fthumbnail.webp&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fcdn.dummyjson.com%2Fproduct-images%2Fbeauty%2Fessence-mascara-lash-princess%2Fthumbnail.webp&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fcdn.dummyjson.com%2Fproduct-images%2Fbeauty%2Fessence-mascara-lash-princess%2Fthumbnail.webp&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
      <span
        class="css-4c00nv"
      />
    </div>
    <div
      class="swiper-slide css-1f8sh1y"
      data-swiper-slide-index="1"
    >
      <img
        alt="image alt"
        class="css-g98gbd"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=https%3A%2F%2Fcdn.dummyjson.com%2Fproduct-images%2Ffragrances%2Fchanel-coco-noir-eau-de%2Fthumbnail.webp&w=3840&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fcdn.dummyjson.com%2Fproduct-images%2Ffragrances%2Fchanel-coco-noir-eau-de%2Fthumbnail.webp&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fcdn.dummyjson.com%2Fproduct-images%2Ffragrances%2Fchanel-coco-noir-eau-de%2Fthumbnail.webp&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fcdn.dummyjson.com%2Fproduct-images%2Ffragrances%2Fchanel-coco-noir-eau-de%2Fthumbnail.webp&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fcdn.dummyjson.com%2Fproduct-images%2Ffragrances%2Fchanel-coco-noir-eau-de%2Fthumbnail.webp&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fcdn.dummyjson.com%2Fproduct-images%2Ffragrances%2Fchanel-coco-noir-eau-de%2Fthumbnail.webp&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fcdn.dummyjson.com%2Fproduct-images%2Ffragrances%2Fchanel-coco-noir-eau-de%2Fthumbnail.webp&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fcdn.dummyjson.com%2Fproduct-images%2Ffragrances%2Fchanel-coco-noir-eau-de%2Fthumbnail.webp&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fcdn.dummyjson.com%2Fproduct-images%2Ffragrances%2Fchanel-coco-noir-eau-de%2Fthumbnail.webp&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
      <span
        class="css-4c00nv"
      />
    </div>
  </div>
  <div
    class="swiper-pagination swiper-pagination-bullets swiper-pagination-horizontal"
  >
    <span
      class="swiper-pagination-bullet swiper-pagination-bullet-active"
    />
    <span
      class="swiper-pagination-bullet"
    />
  </div>
</div>
`;
